File: typescript/packages/core/src/__tests__/tool-invocation/test-utils.ts
import { z } from "zod";
import { PluginBase } from "../../classes/PluginBase";
import { createToolParameters } from "../../utils/createToolParameters";

// Utility to create parameters class
export function createMockParameters(schema: z.ZodTypeAny) {
    return class extends createToolParameters(schema) {};
}

// Create a mock plugin with the given service
export function createMockPlugin(name: string, service: object) {
    class MockPlugin extends PluginBase {
        supportsChain() {
            return true;
        }
    }

    return new MockPlugin(name, [service]);
}

File: python/src/goat-sdk/tests/tool_invocation/test_utils.py
import inspect
from typing import Dict, Any, Type
from pydantic import BaseModel
from goat.classes.plugin_base import PluginBase
    
# Create a mock plugin
def create_mock_plugin(name: str, service: Any) -> PluginBase:
    class MockPlugin(PluginBase):
        def supports_chain(self, chain):
            return True
            
    return MockPlugin(name, [service])
    
# Helper to get the tool from a service class
def get_tool_from_service(service_instance, tool_name: str):
    for name, method in inspect.getmembers(service_instance, inspect.ismethod):
        if hasattr(method, "__goat_tool__") and method.__goat_tool__.name == tool_name:
            return method
    return None
